plugins/CoreUpdater/Updater.php

```diff
@@ -166,41 +157,53 @@ public function oneClickUpdatePartTwo($newVersion = null)
+
+                foreach ($pluginsWithUpdate as $pluginWithUpdate) {
+                    $pluginName      = $pluginWithUpdate['name'];
+                    $messages[]      = $this->translator->translate(
+                        'CoreUpdater_UpdatingPluginXToVersionY',
+                        [$pluginName, $pluginWithUpdate['version']]
+                    );
+                    $pluginInstaller = new PluginInstaller($marketplaceClient);
+                    $pluginInstaller->installOrUpdatePluginFromMarketplace($pluginName);
+                }
+            } catch (MarketplaceApi\Exception $e) {
+                // there is a problem with the connection to the server, ignore for now
+            } catch (Exception $e) {
+                throw new UpdaterException($e, $messages);
+            }
         }
 
-        // we also need to make sure to create a new instance here as otherwise we would change the "global"
-        // environment, but we only want to change piwik version temporarily for this task here
-        $environment = StaticContainer::getContainer()->make('Piwik\Plugins\Marketplace\Environment');
-        $environment->setPiwikVersion($newVersion);
-        /** @var \Piwik\Plugins\Marketplace\Api\Client $marketplaceClient */
-        $marketplaceClient = StaticContainer::getContainer()->make('Piwik\Plugins\Marketplace\Api\Client', array(
-            'environment' => $environment
-        ));
-
         try {
-            $messages[] = $this->translator->translate('CoreUpdater_CheckingForPluginUpdates');
-            $pluginManager = PluginManager::getInstance();
-            $pluginManager->loadAllPluginsAndGetTheirInfo();
-            $loadedPlugins = $pluginManager->getLoadedPlugins();
-
-            $marketplaceClient->clearAllCacheEntries();
-            $pluginsWithUpdate = $marketplaceClient->checkUpdates($loadedPlugins);
-
-            foreach ($pluginsWithUpdate as $pluginWithUpdate) {
-                $pluginName = $pluginWithUpdate['name'];
-                $messages[] = $this->translator->translate('CoreUpdater_UpdatingPluginXToVersionY',
-                    array($pluginName, $pluginWithUpdate['version']));
-                $pluginInstaller = new PluginInstaller($marketplaceClient);
-                $pluginInstaller->installOrUpdatePluginFromMarketplace($pluginName);
+            // incompatible plugins may have already been disabled in oneClickUpdatePartTwo
+            // if this might have failed we try it here again.
+            $disabledPluginNames = $this->disableIncompatiblePlugins($newVersion);
+            if (!empty($disabledPluginNames)) {
+                $messages[] = $this->translator->translate('CoreUpdater_DisablingIncompatiblePlugins', implode(', ', $disabledPluginNames));
             }
-        } catch (MarketplaceApi\Exception $e) {
-            // there is a problem with the connection to the server, ignore for now
-        } catch (Exception $e) {
+        } catch (\Throwable $e) {
```

Comment:
This was `Exception` before, now it's `\Throwable`, I guess no functional difference, just curious if it needs changing?

Reply:
I thought it wouldn't hurt to catch any type of error that might occur while deactivating a plugin. Should be better to show possible php errors or similar as a message instead of running into a fatal.